examples/pic/pic.cpp

```diff
@@ -0,0 +1,637 @@
+/*
+ * main.cpp
+ *
+ *  Created on: Aug 20, 2020
+ *      Author: Jiri Vyskocil
+ */
+#define _USE_MATH_DEFINES
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+#include <chrono>
+#include <cmath>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <llama/llama.hpp>
+#include <random>
+#include <vector>
+
+namespace alpaka
+{
+    template<size_t I, typename TDim, typename TVal>
+    ALPAKA_FN_HOST_ACC auto get(Vec<TDim, TVal> v)
+    {
+        return v[I];
+    }
+} // namespace alpaka
+
+namespace std
+{
+    template<typename TDim, typename TVal>
+    struct tuple_size<alpaka::Vec<TDim, TVal>> : integral_constant<size_t, TDim::value>
+    {
+    };
+
+    template<size_t I, typename TDim, typename TVal>
+    struct tuple_element<I, alpaka::Vec<TDim, TVal>>
+    {
+        using type = TVal;
+    };
+} // namespace std
+
+using Real = double;
+
+// simulation parameters
+constexpr size_t X_ = 1000;
+constexpr size_t Y_ = 1000;
+constexpr Real dx = 0.01;
+constexpr Real dt = 0.99 * dx / M_SQRT2;
+constexpr int ppc = 10;
+constexpr int R_ = 5;
+constexpr int L = 20;
+constexpr Real density = 0.2;
+constexpr Real B0x = 1; // TODO: input.txt had 0
+constexpr Real B0y = 0.5; // TODO: input.txt had 0
+constexpr Real B0z = 0.1; // TODO: input.txt had 0
+constexpr Real v_drift = 0.2;
+constexpr Real v_therm = 0.005;
+constexpr Real alpha = 0.8;
+constexpr int NSTEPS = 2000;
+constexpr int outputInterval = 10;
+constexpr Real Rcx = R_ * dx + 0;
+constexpr Real Rcy = R_ * dx + 0;
+constexpr Real Rcz = 0;
+constexpr auto reportInterval = 10;
```

Comment:
@sliwowitz could you please cross check these parameters if they are correct? I also would like to have a more heavy simulation, with around 5x to 10x more work. I also think we might have too little particles. There are only 2000 particles on a 1000x1000 grid, which is the reason why the particle kernels have almost no impact on the runtime. How much can we increase the particle count? What is a realistic scenario? Thx!

Reply:
Discussed offline.